test/adapters/test_middleware_test.rb

```diff
@@ -81,7 +94,7 @@ def test_middleware_allow_different_outcomes_for_the_same_request
     end
 
     def test_yields_env_to_stubs
-      @stubs.get '/hello' do |env|
+      @stubs.get 'http://foo.com/hello' do |env|
```

Comment:
See comment above regarding the host.
We still want to allow people to stub paths (hence URLs with no host).
We should then keep this tests unchanged and duplicate all of them to test that the stub works WITH and WITHOUT the host specified

Reply:
Yeah, I'm actually kind of confused about this. The question is, what does it mean when user doesn't specify any host in the stub? Does it mean the stub should be available for all hosts? I mean, should `connection.get('http://domain.com/path')` look for stubs declared like `stubs.get('/path')` as well as `stubs.get('http://domain.com/path')`.

I think the use case mentioned in the issue (https://github.com/lostisland/faraday/issues/358#issue-30252173) is trying to make sure Faraday is hitting the right domain. With my changes is still possible to make stubs with no host, but they'll only match requests with no host as well. Is this wrong?

Reply:
Apologies for not being more clear from the beginning.
The wanted behaviour is the following:

1. If a hostname is specified on the stub, then both the hostname AND the path should match.
2. If no hostname is specified on the stub (hence only the path), then any hostname should match.

Point 1 will "fix" #358, while point 2 will maintain backward compatibility, as the current behaviour is "stub this path on any host" and we should keep that working as well 👍